### PR TITLE
Change ClassCleanup default behavior to be "EndOfClass"

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/1316

### CODE REVIEW

File: src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs [hunk 1]
```diff
@@ -9,12 +9,12 @@ namespace Microsoft.VisualStudio.TestTools.UnitTesting;
 public enum ClassCleanupBehavior
 {
     /// <summary>
-    /// Run at end of assembly.
+    /// Run at end of class.
     /// </summary>
-    EndOfAssembly,
```
Comment (Evangelink) [inline src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs:14]: Is this change required? If so can you explain why?

File: src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 ﻿#nullable enable
+Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestClassInfo.RunClassCleanup(Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior classCleanupLifecycle = Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior.EndOfClass) -> string?
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Unshipped.txt]: I would keep this one in shipped because it's an already shipped API.

File: src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupExecutionAttribute.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ public class ClassCleanupExecutionAttribute : Attribute
     /// <summary>
     /// Default class cleanup execution.
     /// </summary>
-    public static readonly ClassCleanupBehavior DefaultClassCleanupLifecycle = ClassCleanupBehavior.EndOfAssembly;
+    public static readonly ClassCleanupBehavior DefaultClassCleanupLifecycle = ClassCleanupBehavior.EndOfClass;
```
Comment (Evangelink) [inline src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupExecutionAttribute.cs]: Could you check in AzDo and GitHub if you see some usage of this field? It seems to be unused inside our framework so maybe it could be removed.

File: test/IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests/SuiteLifeCycleTests.cs [hunk 4]
```diff
@@ -167,6 +167,7 @@ private void ValidateTestRunLifecycle(string targetFramework)
             {(targetFramework == "net6.0"
                 ? "Console: LifeCycleClassCleanup.DisposeAsync was called\r\nConsole: LifeCycleClassCleanup.Dispose was called"
                 : "Console: LifeCycleClassCleanup.Dispose was called")}
+            Console: LifeCycleClassCleanup.ClassCleanup was called
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests/SuiteLifeCycleTests.cs:171]: This seems to indicate that some cleanup was changed, is it expected? If so, please try to help review by adding some comments helping to understand the changes so that reviewer isn't forced to check it up.

File: test/UnitTests/MSTestAdapter.UnitTests/Execution/TestClassInfoTests.cs [hunk 5]
```diff
@@ -491,7 +491,7 @@ public void RunClassCleanupShouldReturnAssertFailureExceptionDetails()
         _testClassInfo.ClassCleanupMethod = typeof(DummyTestClass).GetMethod(nameof(DummyTestClass.ClassCleanupMethod));
 
         // Act
-        var classCleanup = _testClassInfo.RunClassCleanup();
+        var classCleanup = _testClassInfo.RunClassCleanup(UTF.ClassCleanupBehavior.EndOfAssembly);
```
Comment (Evangelink) [inline test/UnitTests/MSTestAdapter.UnitTests/Execution/TestClassInfoTests.cs:494]: Does it need to be changed? If it does, are we having some equivalent test for the other beahviour?

Same comment for the other tests.

File: src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs (hunk 1 above)
Comment (engyebrahim) [reply src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs:14]: Yes required because the default value for any new Enum is zero and we need to make sure that the default is always the `EndOfClass`

File: test/UnitTests/MSTestAdapter.UnitTests/Execution/TestClassInfoTests.cs (hunk 5 above)
Comment (engyebrahim) [reply test/UnitTests/MSTestAdapter.UnitTests/Execution/TestClassInfoTests.cs:494]: yes because it was initially testing the `EndOfAssembly` behavior and thrown this type of exception is in it's flow not in the `endofclass's` flow. 

and yeah there is tests many now tests the `endofclass` behavior as it's the default.

File: test/IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests/SuiteLifeCycleTests.cs (hunk 4 above)
Comment (engyebrahim) [reply test/IntegrationTests/MSTest.VstestConsoleWrapper.IntegrationTests/SuiteLifeCycleTests.cs:171]: yes changed because because this class has the default behavior which is `EndOfClass` now and before was `EndOfAssembly`

File: src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Shipped.txt [hunk 6]
```diff
@@ -15,6 +15,7 @@ Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.LogMessageListe
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.LogMessageListener.StandardError.get -> string!
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.LogMessageListener.StandardOutput.get -> string!
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestAssemblyInfo
+Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution.TestClassInfo.RunClassCleanup(Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior classCleanupLifecycle = Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior.EndOfClass) -> string?
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Shipped.txt]: Is the file still sorted alphabetically? If not, could you please update it?

File: src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs (hunk 1 above)
Comment (Evangelink) [reply src/TestFramework/TestFramework/Attributes/Lifecycle/Cleanup/ClassCleanupBehavior.cs:14]: It's probably not so important for our case but let's do it anyways. Thanks for the explanation :)

File: src/TestFramework/TestFramework/PublicAPI/PublicAPI.Unshipped.txt [hunk 7]
```diff
@@ -1,4 +1,6 @@
 ﻿#nullable enable
+Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior.EndOfAssembly = 1 -> Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupBehavior
```
Comment (Evangelink) [inline src/TestFramework/TestFramework/PublicAPI/PublicAPI.Unshipped.txt]: Sorry my previous comment wasn't super clear. These changes should also stay in the shipped. Basically, any change of public API already shipped should stay in shipped.

**Review by Evangelink — COMMENTED** [review]
A few more comments and we are good to go. Thanks for the work @engyebrahim

**Review by Evangelink — APPROVED** [review]
